// format-code/esm5/lib/format-code.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { Component, Input, Renderer2, ElementRef } from "@angular/core";
var FormatCodeComponent = /** @class */ (function() {
  function FormatCodeComponent(renderer, el) {
    this.renderer = renderer;
    this.el = el;
    this.color = "rgb(42,100,250)";
    this.bgColor = "rgba(0,0,30,0.9)";
    this.font = "consolas, monospace";
  }
  /**
   * @return {?}
   */
  FormatCodeComponent.prototype.ngOnInit
  /**
   * @return {?}
   */ = function() {
    this.renderer.setStyle(
      this.el.nativeElement.firstElementChild,
      "color",
      this.color
    );
    this.renderer.setStyle(
      this.el.nativeElement.firstElementChild,
      "backgroundColor",
      this.bgColor
    );
    this.renderer.setStyle(
      this.el.nativeElement.firstElementChild,
      "font-family",
      this.font
    );
  };
  FormatCodeComponent.decorators = [
    {
      type: Component,
      args: [
        {
          selector: "format-code",
          template:
            "\n    <div>\n      <ng-content></ng-content>\n    </div>\n  ",
          styles: [
            "\n    div {\n      padding: 20px;\n      margin: 10px auto;\n      box-shadow: 1px 3px 10px 1px rgba(0, 0, 0, 0.3);\n      border-radius: 7px;\n    }\n  "
          ]
        }
      ]
    }
  ];
  /** @nocollapse */
  FormatCodeComponent.ctorParameters = function() {
    return [{ type: Renderer2 }, { type: ElementRef }];
  };
  FormatCodeComponent.propDecorators = {
    color: [{ type: Input }],
    bgColor: [{ type: Input }],
    font: [{ type: Input }]
  };
  return FormatCodeComponent;
})();
export { FormatCodeComponent };
if (false) {
  /** @type {?} */
  FormatCodeComponent.prototype.color;
  /** @type {?} */
  FormatCodeComponent.prototype.bgColor;
  /** @type {?} */
  FormatCodeComponent.prototype.font;
  /** @type {?} */
  FormatCodeComponent.prototype.renderer;
  /** @type {?} */
  FormatCodeComponent.prototype.el;
}
