// format-code/bundles/format-code.umd.js
(function(global, factory) {
  typeof exports === "object" && typeof module !== "undefined"
    ? factory(exports, require("@angular/core"))
    : typeof define === "function" && define.amd
    ? define("format-code", ["exports", "@angular/core"], factory)
    : factory((global["format-code"] = {}), global.ng.core);
})(this, function(exports, core) {
  "use strict";

  /**
   * @fileoverview added by tsickle
   * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
   */
  var FormatCodeComponent = /** @class */ (function() {
    function FormatCodeComponent(renderer, el) {
      this.renderer = renderer;
      this.el = el;
      this.color = "rgb(42,100,250)";
      this.bgColor = "rgba(0,0,30,0.9)";
      this.font = "consolas, monospace";
    }
    /**
     * @return {?}
     */
    FormatCodeComponent.prototype.ngOnInit
    /**
     * @return {?}
     */ = function() {
      this.renderer.setStyle(
        this.el.nativeElement.firstElementChild,
        "color",
        this.color
      );
      this.renderer.setStyle(
        this.el.nativeElement.firstElementChild,
        "backgroundColor",
        this.bgColor
      );
      this.renderer.setStyle(
        this.el.nativeElement.firstElementChild,
        "font-family",
        this.font
      );
    };
    FormatCodeComponent.decorators = [
      {
        type: core.Component,
        args: [
          {
            selector: "format-code",
            template:
              "\n    <div>\n      <ng-content></ng-content>\n    </div>\n  ",
            styles: [
              "\n    div {\n      padding: 20px;\n      margin: 10px auto;\n      box-shadow: 1px 3px 10px 1px rgba(0, 0, 0, 0.3);\n      border-radius: 7px;\n    }\n  "
            ]
          }
        ]
      }
    ];
    /** @nocollapse */
    FormatCodeComponent.ctorParameters = function() {
      return [{ type: core.Renderer2 }, { type: core.ElementRef }];
    };
    FormatCodeComponent.propDecorators = {
      color: [{ type: core.Input }],
      bgColor: [{ type: core.Input }],
      font: [{ type: core.Input }]
    };
    return FormatCodeComponent;
  })();

  /**
   * @fileoverview added by tsickle
   * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
   */
  var FormatCodeModule = /** @class */ (function() {
    function FormatCodeModule() {}
    FormatCodeModule.decorators = [
      {
        type: core.NgModule,
        args: [
          {
            declarations: [FormatCodeComponent],
            imports: [],
            exports: [FormatCodeComponent]
          }
        ]
      }
    ];
    return FormatCodeModule;
  })();

  /**
   * @fileoverview added by tsickle
   * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
   */

  /**
   * @fileoverview added by tsickle
   * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
   */

  exports.FormatCodeComponent = FormatCodeComponent;
  exports.FormatCodeModule = FormatCodeModule;

  Object.defineProperty(exports, "__esModule", { value: true });
});
